app/backend/core/confighelper.py

```
@@ -40,10 +44,63 @@ def _get_model_config(self, index):
+    
+    def _get_quick_prompt(self, assistant, index):
+        prefix = f"BACKEND_COMMUNITY_ASSISTANT_{assistant}_QUICK_PROMPT_{index}_"
+        config = QuickPrompt(
+            label= getenv(f"{prefix}LABEL"),
+            prompt= getenv(f"{prefix}PROMPT"),
+            tooltip= getenv(f"{prefix}TOOLTIP")
+        )
+        return config
+    
+    def _get_quick_prompt_list(self, assistant):
+        quick_prompts = []
+        highest_index = self._find_highest_index(r"BACKEND_COMMUNITY_ASSISTANT_"+str(assistant)+r"_QUICK_PROMPT_(\d+)") 
+        for i in range(1, highest_index + 1):
+            quick_prompts.append(self._get_quick_prompt(assistant,i))
+        return quick_prompts
+
+    def _get_example(self, assistant, index):
+        prefix = f"BACKEND_COMMUNITY_ASSISTANT_{assistant}_EXAMPLES_{index}_"
+        config = ExampleModel(
+            text= getenv(f"{prefix}TEXT"),
+            value= getenv(f"{prefix}VALUE"),
+            system= getenv(f"{prefix}SYSTEM", "")
+        )
+        return config
+    
+    def _get_example_list(self, assistant):
+        examples = []
+        highest_index = self._find_highest_index(r"BACKEND_COMMUNITY_ASSISTANT_"+str(assistant)+r"_EXAMPLES_(\d+)")
+        for i in range(1, highest_index + 1):
+            examples.append(self._get_example(assistant,i))
+        return examples
+
+    def _get_community_assistant(self, index):
+        prefix = f"BACKEND_COMMUNITY_ASSISTANT_{index}_"
+        config = CommunityAssistantConfig(
+            title= getenv(f"{prefix}TITLE"),
+            description= getenv(f"{prefix}DESCRIPTION"),
+            system_message= getenv(f"{prefix}SYSTEM_MESSAGE"),
+            examples= self._get_example_list(index),
+            quick_prompts= self._get_quick_prompt_list(index),
+            temperature= float(getenv(f"{prefix}TEMPERATURE", 0.7)),
+            max_output_tokens= int(getenv(f"{prefix}MAX_TOKENS", 100)),
+            id = str(getenv(f"{prefix}ID", uuid())),
```

I wouldn't provide an a default id with `uuid()` if no id is present in the env. Should throw an error in this case.

Otherwise we generate a new random id every server restart and we will have a lot of duplicated community assistents.

---

Yes, might not be the best idea 😅